rust/blockstore/src/arrow/block/delta/data_record.rs

```diff
@@ -1,34 +1,22 @@
+use super::data_record_size_tracker::DataRecordSizeTracker;
 use super::BlockKeyArrowBuilder;
+use crate::arrow::types::ArrowWriteableValue;
 use crate::{
     arrow::types::ArrowWriteableKey,
     key::{CompositeKey, KeyWrapper},
 };
-use arrow::{
-    array::{
-        Array, ArrayRef, BinaryBuilder, FixedSizeListBuilder, Float32Builder, RecordBatch,
-        StringBuilder, StructArray,
-    },
-    datatypes::{Field, Fields},
-    util::bit_util,
-};
-use chroma_types::{chroma_proto::UpdateMetadata, DataRecord};
+use arrow::{array::RecordBatch, util::bit_util};
+use chroma_types::DataRecord;
 use parking_lot::RwLock;
-use prost::Message;
 use std::{collections::BTreeMap, sync::Arc};
 
-// Convenience type for the storage entry
-// (id, embedding, metadata, document)
-type DataRecordStorageEntry = (String, Vec<f32>, Option<Vec<u8>>, Option<String>);
-
 #[derive(Debug)]
 struct Inner {
-    storage: BTreeMap<CompositeKey, DataRecordStorageEntry>,
-    prefix_size: usize,
-    key_size: usize,
-    id_size: usize,
-    embedding_size: usize,
-    metadata_size: usize,
-    document_size: usize,
+    storage: BTreeMap<
+        CompositeKey,
+        <&'static chroma_types::DataRecord<'static> as ArrowWriteableValue>::PreparedValue,
```

Comment:
The need for static refs here seems weird to me?

Reply:
I just need to get at the `PreparedValue` type, from my understanding I don't think 'static here should constrain anything? otherwise would have to introduce a lifetime parameter that's not actually used

Reply:
I am asking

&'static chroma_types::DataRecord<'static>

vs 
chroma_types::DataRecord<'static>

Reply:
because `ArrowWritableValue` is implemented for `&DataRecord`, not `DataRecord`